src/libs/models/BankAccount.ts

```diff
@@ -0,0 +1,241 @@
+     * @private
+     */
+    private getState(): State | undefined {
+        return this.json.accountData?.state;
+    }
+
+    /**
+     * @returns - if the bank account is open.
+     */
+    isOpen(): boolean {
+        return this.getState() === BankAccount.STATE.OPEN;
+    }
+
+    /**
+     * @deprecated Use !isPending instead.
+     * @returns - if the bank account is verified.
+     */
+    isVerified(): boolean {
+        return !this.isPending();
+    }
+
+    /**
+     * @returns - if he user still needs to enter the 3 micro deposit amounts.
+     */
+    isPending(): boolean {
+        return this.getState() === BankAccount.STATE.PENDING;
+    }
+
+    /**
+     * @returns - if success team is currently verifying the bank account data provided by the user.
+     */
+    isVerifying(): boolean {
+        return this.getState() === BankAccount.STATE.VERIFYING;
+    }
+
+    /**
+     * @returns - if the user didn't finish entering all their info.
+     */
+    isInSetup(): boolean {
+        return this.getState() === BankAccount.STATE.SETUP;
+    }
+
+    /**
+     * @returns - if the bank account is locked.
+     */
+    isLocked(): boolean {
+        return this.getState() === BankAccount.STATE.LOCKED;
+    }
+
+    /**
+     * @returns - if the account is the default credit account.
+     */
+    isDefaultCredit(): boolean | undefined {
+        return this.json.accountData?.defaultCredit;
+    }
+
+    /**
+     * @returns - if the account can be used for paying other people.
+     */
+    isWithdrawal(): boolean | undefined {
+        return this.json.accountData?.allowDebit;
+    }
+
+    /**
+     * @NOTE WARNING KEEP IN SYNC WITH THE PHP
+     * @returns - client ID of the bank account.
+     */
+    getClientID() {
+        // eslint-disable-next-line max-len
+        return `${Str.makeID(this.getMaskedAccountNumber() || "")}${Str.makeID(this.getAddressName() || "")}${Str.makeID(this.getRoutingNumber() || "")}${this.getType()}`;
+    }
+
+    /**
+     * @returns - type of the bank account.
+     * @private
+     */
+    private getType(): BankAccountType {
+        return this.isWithdrawal() ? 'withdrawal' : 'direct-deposit';
+    }
+
+    /**
+     * @returns - Return the internal json data structure used by auth.
+     */
+    getJSON(): BankAccountJSON {
+        return this.json;
+    }
+
+    /**
+     * @returns - Whether or not this bank account has been risk checked
+     */
+    isRiskChecked(): boolean | undefined {
+        return this.json.accountData?.riskChecked;
+    }
+
+    /**
+     * @returns - date when the 3 micro amounts for validation were supposed to reach the bank account.
+     */
+    getValidateCodeExpectedDate(): string {
+        return this.json.validateCodeExpectedDate || '';
+    }
+
+    /**
+     * @returns - country of the bank account.
+     */
+    getCountry(): string {
+        return this.json.accountData?.additionalData?.country as string || CONST.COUNTRY.US;
```

Comment:
I think this way we could avoid force-casting with `as string`:

```suggestion
        return this.json.accountData?.additionalData?.country ?? CONST.COUNTRY.US;
```

Then, do it for all cases below